pkg/api/types.go

```diff
@@ -136,15 +153,24 @@ type InstanceSpec struct {
 }
 
 type InstanceStatus struct {
-	State InstanceState
+	Conditions []InstanceCondition
```

Comment:
Does the size of this slice vary or will it always match the number of InstanceConditions we have defined?
Also, should it be a map instead of slice so its easier to find the particular condition someone is looking for?